### PR TITLE
Fix development makefile to allow compilation.

### PR DESCRIPTION
**Pull request by jeromekelleher**
The suppressed error is something that has to happen in the Python C Api

### CODE REVIEW

**Comment by jeromekelleher** [comment]
I just want to make sure I haven't mucked up the github action here @benjeffery :eyes:

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -25,8 +25,6 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - uses: actions/setup-python@v2
-    - name: set PY
-      run: echo "::set-env name=PY::$(python -c 'import hashlib, sys;print(hashlib.sha256(sys.version.encode()+sys.executable.encode()).hexdigest())')"
     - uses: actions/cache@v1
```
Comment (benjeffery) [inline .github/workflows/lint.yml:28]: While you're here you might as well bump cache to v2

**Review by benjeffery — APPROVED** [review]

**Comment by benjeffery** [comment]
Yep looks good!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (jeromekelleher) [reply .github/workflows/lint.yml]: Done

**Comment by jeromekelleher** [comment]
Merging manually as this changes github actions.

**Pull request merged**